### ISSUE TITLE
Logging prints one character on each line

### ISSUE THREAD

**Issue by Foorack** [GlowstoneMC/Glowstone#915]
Not sure if this is a Glowstone, BanManager or Hikari bug. :thinking: Posting it here as it works fine with Spigot/Paper. Running Glowstone version https://github.com/GlowstoneMC/Glowstone/commit/7ff7d959f466a4702237f9f9931099bbd5eb0d0e

```
2018/05/05 21:44:10 [INFO] Enabling BanManager v5.15.0-SNAPSHOT
2018/05/05 21:44:10 [WARNING] 2
2018/05/05 21:44:10 [WARNING] 1
2018/05/05 21:44:10 [WARNING] 9
2018/05/05 21:44:10 [WARNING]  
2018/05/05 21:44:10 [WARNING] [
2018/05/05 21:44:10 [WARNING] m
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] i
2018/05/05 21:44:10 [WARNING] n
2018/05/05 21:44:10 [WARNING] ]
2018/05/05 21:44:10 [WARNING]  
2018/05/05 21:44:10 [WARNING] I
2018/05/05 21:44:10 [WARNING] N
2018/05/05 21:44:10 [WARNING] F
2018/05/05 21:44:10 [WARNING] O
2018/05/05 21:44:10 [WARNING]  
2018/05/05 21:44:10 [WARNING] m
2018/05/05 21:44:10 [WARNING] e
2018/05/05 21:44:10 [WARNING] .
2018/05/05 21:44:10 [WARNING] c
2018/05/05 21:44:10 [WARNING] o
2018/05/05 21:44:10 [WARNING] n
2018/05/05 21:44:10 [WARNING] f
2018/05/05 21:44:10 [WARNING] u
2018/05/05 21:44:10 [WARNING] s
2018/05/05 21:44:10 [WARNING] e
2018/05/05 21:44:10 [WARNING] r
2018/05/05 21:44:10 [WARNING] .
2018/05/05 21:44:10 [WARNING] b
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] n
2018/05/05 21:44:10 [WARNING] m
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] n
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] g
2018/05/05 21:44:10 [WARNING] e
2018/05/05 21:44:10 [WARNING] r
2018/05/05 21:44:10 [WARNING] .
2018/05/05 21:44:10 [WARNING] i
2018/05/05 21:44:10 [WARNING] n
2018/05/05 21:44:10 [WARNING] t
2018/05/05 21:44:10 [WARNING] e
2018/05/05 21:44:10 [WARNING] r
2018/05/05 21:44:10 [WARNING] n
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] l
2018/05/05 21:44:10 [WARNING] .
2018/05/05 21:44:10 [WARNING] h
2018/05/05 21:44:10 [WARNING] i
2018/05/05 21:44:10 [WARNING] k
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] r
2018/05/05 21:44:10 [WARNING] i
2018/05/05 21:44:10 [WARNING] .
2018/05/05 21:44:10 [WARNING] H
2018/05/05 21:44:10 [WARNING] i
2018/05/05 21:44:10 [WARNING] k
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] r
2018/05/05 21:44:10 [WARNING] i
2018/05/05 21:44:10 [WARNING] D
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] t
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] S
2018/05/05 21:44:10 [WARNING] o
2018/05/05 21:44:10 [WARNING] u
2018/05/05 21:44:10 [WARNING] r
2018/05/05 21:44:10 [WARNING] c
2018/05/05 21:44:10 [WARNING] e
2018/05/05 21:44:10 [WARNING]  
2018/05/05 21:44:10 [WARNING] -
2018/05/05 21:44:10 [WARNING]  
2018/05/05 21:44:10 [WARNING] b
2018/05/05 21:44:10 [WARNING] m
2018/05/05 21:44:10 [WARNING] -
2018/05/05 21:44:10 [WARNING] l
2018/05/05 21:44:10 [WARNING] o
2018/05/05 21:44:10 [WARNING] c
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] l
2018/05/05 21:44:10 [WARNING]  
2018/05/05 21:44:10 [WARNING] -
2018/05/05 21:44:10 [WARNING]  
2018/05/05 21:44:10 [WARNING] S
2018/05/05 21:44:10 [WARNING] t
2018/05/05 21:44:10 [WARNING] a
2018/05/05 21:44:10 [WARNING] r
2018/05/05 21:44:10 [WARNING] t
2018/05/05 21:44:10 [WARNING] e
2018/05/05 21:44:10 [WARNING] d
2018/05/05 21:44:10 [WARNING] .
2018/05/05 21:44:11 [INFO] Loaded 327 bans into memory
2018/05/05 21:44:11 [INFO] Loaded 0 mutes into memory
2018/05/05 21:44:11 [INFO] Loaded 38 ip bans into memory
2018/05/05 21:44:11 [INFO] Loaded 0 ip mutes into memory
2018/05/05 21:44:11 [INFO] Loaded 0 ip range bans into memory
2018/05/05 21:44:11 [INFO] Loaded 15 name bans into memory```

**Comment by Pr0methean** [comment]
Pretty sure it's not a Glowstone bug. Since it includes the prefix "[main] INFO", it must be another logging framework incorrectly feeding into slf4j when they should be running side-by-side. What logging framework do BanManager and Hikari use?

**Comment by mikroskeem** [comment]
Hikari uses slf4j, no idea about BanManager

**Comment by aramperes** [comment]
The logged [line](https://github.com/brettwooldridge/HikariCP/blob/HikariCP-2.4.7/src/main/java/com/zaxxer/hikari/HikariDataSource.java#L70) comes from HikariCP. BanManager uses HikariCP v2.4.7 (which is almost 2 years old). Anyhow, the `LOGGER` field is initialized by slf4j's LoggerFactory (v1.7.21 in this case).

I think it might be a bug on our end. Perhaps we aren't handing slf4j correctly (or at all?) with log4j.

**Comment by aki-ks** [comment]
I remember that i've got the same issue if I use `System.out.println()` on linux.

**Comment by aki-ks** [comment]
I think there was something wrong with the [`LoggerOutputStream`](https://github.com/GlowstoneMC/Glowstone/blob/a12eebbf8aad46f4d46ff4af86c59a55a39ad49e/src/main/java/net/glowstone/ConsoleManager.java#L219-L238) implementation. It flushed after every received byte/character.

**Comment by Pr0methean** [comment]
I can reproduce this in Xubuntu 18.04 on JDK10, but not JDK8.

**Comment by Pr0methean** [comment]
The problem isn't with LoggerOutputStream, it's that the PrintStream instances we wrap around them as System.err and System.out had autoFlush on, and someone assumed System.err and System.out would not. Should be fixed by 8241f28; can anyone confirm?

**Comment by aramperes** [comment]
Appears to work on the latest build (`2018.10.0-SNAPSHOT.57c7b28`). Tested on Ubuntu 16.04.

```
03:11:47 [INFO] Enabling BanManager v5.15.0
03:11:47 [WARNING] 64 [main] INFO me.confuser.banmanager.internal.hikari.HikariDataSource - bm-local - Started.
```

Please let us know if this is not resolved on your end.
